keystorage/yubihsm2/src/test/java/tech/pegasys/signers/yubihsm/YubiHsmSessionTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.yubihsm;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import tech.pegasys.signers.yubihsm.backend.YubiHsmBackend;
+import tech.pegasys.signers.yubihsm.backend.YubiHsmBackendTestFactory;
+import tech.pegasys.signers.yubihsm.exceptions.YubiHsmException;
+import tech.pegasys.signers.yubihsm.model.Opaque;
+
+import java.nio.file.Path;
+
+import io.specto.hoverfly.junit5.HoverflyExtension;
+import io.specto.hoverfly.junit5.api.HoverflyConfig;
+import io.specto.hoverfly.junit5.api.HoverflySimulate;
+import org.apache.tuweni.bytes.Bytes;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+@Nested
+@HoverflySimulate(
+    enableAutoCapture = true,
+    config =
+        @HoverflyConfig(captureAllHeaders = true, proxyLocalHost = true, statefulCapture = true))
```

Comment:
Does the capture properties being set here mean it is capturing data to the hoverfly file? I would of expected to not see these set now that we have the data in the hoverfly capture file.

Reply:
these config allows to capture the payload on first access, and then use captured payload in subsequent access. I've removed the unnecessary bits as we have captured everything that is required.